### PR TITLE
[tests] Adding tests for captioning losses and metrics

### PR DESCRIPTION
**Pull request by vedanuj**
Adding tests for caption losses and metrics. For issue #80 

Loss : `CaptionCrossEntropyLoss `
Metric : `CaptionBleu4Metric `

### CODE REVIEW

File: tests/modules/vocab.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+road
+riding
+motor
+bike
+the
+countryside
+back
+of
+motorcycle
+<unk>
```
Comment (apsdehal) [inline tests/modules/vocab.txt]: Normally, <unk> is the first word in vocab.

**Comment by vedanuj** [comment]
Added tests for `CaptionProcessor`. For `BeamSearch` it will need a bit of refactoring to be able to write the tests easily. I will do that together.

File: tests/modules/vocab.txt (hunk 1 above)
Comment (vedanuj) [reply tests/modules/vocab.txt]: Actually upon checking we need not add the `<unk>` token as `BaseVocab` class adds this token. Removed this.

**Comment by apsdehal** [comment]
For beam search, are you going to do that in a separate PR?

**Pull request merged**